rskj-core/src/test/java/co/rsk/peg/bitcoin/BitcoinTestUtils.java

```diff
@@ -154,4 +155,23 @@ private static void signLegacyTransactionInputFromP2shMultiSig(BtcTransaction tr
             input.setScriptSig(inputScriptSig);
         }
     }
+
+    public static List<Sha256Hash> generateTransactionInputsSigHashes(BtcTransaction btcTx) {
+        List<Sha256Hash> sigHashes = new ArrayList<>();
+        List<TransactionInput> inputs = btcTx.getInputs();
+        for (TransactionInput input : inputs) {
+            Sha256Hash sigHash = generateSigHashForP2SHTransactionInput(btcTx, inputs.indexOf(input));
+            sigHashes.add(sigHash);
+        }
+        return sigHashes;
```

Comment:
```
return Stream.iterate(0, i -> i + 1)
            .limit(btcTx.getInputs().size())
            .map(i -> generateSigHashForP2SHTransactionInput(btcTx, i))
            .collect(Collectors.toList());
            
```

This is a more functional version. Wdyt?

Reply:
udpated